tests/test_emulator.cpp

```diff
@@ -200,13 +200,58 @@ class EmulatorTest : public testing::Test {
         em_status_t ret = EM_ERROR;
 
         vcpu = vcpu_original;
-        assemble_decode(insn, vcpu.rip, &size, &count, &ret);
-        ASSERT_TRUE(ret != EM_ERROR);
+        assemble_decode(mode, insn, vcpu.rip, &size, &count, &ret);
+        ASSERT_TRUE(ret != EM_ERROR)
```

Comment:
nit: I probably abused `ASSERT_TRUE` and `EXPECT_TRUE` when I last modified this function. Later I realized that the `_EQ` macros would be a better fit for value comparisons, because they would give us more helpful information in case of assertion failure. Please feel free to address this problem in this patch.